HuntHelper/Managers/Hunts/SirenHuntsManager.cs

```diff
@@ -0,0 +1,158 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+
+namespace HuntHelper.Managers.Hunts;
+
+public class SirenHuntsManager
+{
+    private IDictionary<string, SirenHuntsPatchData> _sirenHuntsData;
+
+    private readonly Configuration _config;
+    private readonly IChatGui _chatGui;
+    private readonly MapDataManager _mapDataManager;
+    private readonly string _sirenHuntsFilePath;
+
+    public SirenHuntsManager(
+        Configuration config,
+        IChatGui chatGui,
+        MapDataManager mapDataManager,
+        string sirenHuntsFilePath)
+    {
+        _config = config;
+        _chatGui = chatGui;
+        _mapDataManager = mapDataManager;
+        _sirenHuntsFilePath = sirenHuntsFilePath;
+        LoadSirenHuntsData();
+    }
+
+    public string GetSirenHuntsLink(IList<HuntTrainMob> mobPositions)
+    {
+        var link = GenerateLink(mobPositions);
+        SendLinkToChat(link);
+        return link;
+    }
+
+    private void SendLinkToChat(string link)
+    {
+        var linkMessage = new SeStringBuilder();
+        linkMessage.AddUiForeground(67);
+        linkMessage.AddText("[Hunt Helper] ");
+        linkMessage.AddUiForegroundOff();
+
+        if (link.Length == 0)
+        {
+            linkMessage.AddUiForeground(31);
+            linkMessage.AddText("No link generated");
+            linkMessage.AddUiForegroundOff();
+        }
+        else
+        {
+            linkMessage.AddUiForeground(58);
+            linkMessage.AddText("Generated link: ");
+            linkMessage.AddUiForeground(33);
+            linkMessage.AddText(link);
+            linkMessage.AddUiForegroundOff();
+        }
+
+        _chatGui.Print(linkMessage.BuiltString);
+    }
+
+    public string GenerateLink(IList<HuntTrainMob> mobPositions)
+    {
+        var pathSegments = new List<string>
+            {
+                GetUrlComponent("HW", mobPositions),
+                GetUrlComponent("SB", mobPositions),
+                GetUrlComponent("SHB", mobPositions),
+                GetUrlComponent("EW", mobPositions)
+            }
+            .Where(component => 0 < component.Length)
+            .ToList();
+
+        if (pathSegments.Count == 0) return string.Empty;
+
+        var marksPath = string.Join('&', pathSegments);
+        var baseUrl = _config.SirenHuntsBaseUrl.TrimEnd('/');
+        return $"{baseUrl}/{marksPath}";
+    }
+
+    private string GetUrlComponent(string patch, IList<HuntTrainMob> mobs)
+    {
+        var patchMobs = _sirenHuntsData[patch].MobOrder.Select(patchOrderMobName =>
+            mobs.FirstOrDefault(mob =>
+            {
+                var mobName = mob.Name;
+                if (0 < mob.Instance) mobName += $" {mob.Instance}";
+                return string.Equals(mobName, patchOrderMobName, StringComparison.CurrentCultureIgnoreCase);
```

Comment:
i forgot about this, but this string comparison i think won't work for other languages? it's comparing the mob names to the json data, which is in english. i think i should replace this with a mob id based comparison.